akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java

```diff
@@ -612,6 +612,19 @@ void groupByExample() {
     // #groupBy
   }
 
+  void groupByWithAsyncExample() {
+    // #groupByWithAsync
+    Source.range(1, 10)
+            .groupBy(2, i -> i % 2 == 0) // create two sub-streams with odd and even numbers
+            .async()
```

Comment:
This is trickily enough not the right thing, `.async()` creates an async boundary between the graph so far and the next steps, to create a separate async island for each substream you would have to do `.via(Flow.reduce(...).async())`

Reply:
ah gotcha! I wasn't aware about this, and maybe I've been using it wrongly. I'll update the example.

Reply:
Just `groupBy().async` would put the graph up to that point, so `Source.range` and `groupBy` grouping on one stream interpreter actor, and then all the downstreams and the substream merge and sink would run in another single actor. So it has impact but it does not do what we want to show here (one separate stream interpreter per substream).

Reply:
@johanandren : does this diagram depict the right understanding ? before making the change I wanted to understand correctly.

![groupby_async](https://github.com/akka/akka/assets/16983076/97ef13f7-6ac2-4a3d-8a6f-9fe1e84b870c)


I think the current example is state A and we want to show state B in the example by adding async to the flow before merging sub streams. correct?

Reply:
Just doing `.async` after groupBy means that you end up with two interpreters/actors running the stream, one up to the `.async` and one after.
<img width="631" alt="Screenshot 2024-01-16 at 13 22 34" src="https://github.com/akka/akka/assets/666915/7c2f69a5-ac31-46f0-a6d2-8d8da0ba9fe7">

State B is correct, `Flow.a.b.b.async` adds a boundary around each materialisation of that flow

Reply:
@johanandren : the image you attached is the current state right ? and State B which is correct will be done by adding async in the example as proposed by you , right?

also the materialisation of the stream happens even between the actors or only at the end of the stream?

Reply:
Thanks for the answer @johanandren.  I've now changed the example as suggested, can you please review it again?